packages/editor/src/components/post-actions/actions.js

```diff
@@ -589,6 +589,10 @@ export function usePostActions( { postType, onActionPerformed, context } ) {
 		},
 		[ postType ]
 	);
+	const { registerPostTypeActions } = unlock( useDispatch( editorStore ) );
+	useEffect( () => {
+		registerPostTypeActions( postType );
+	}, [ registerPostTypeActions, postType ] );
```

Comment:
I don't love these kinds of effects tbh. Not sure what the alternative is, I guess filtered context wouldn't work.

Reply:
This is for me a good thing: Only register the actions and run the filter when needed (when using usePostAction hook), that said, I agree that the code is weird, but I don't know of another pattern that provides the same benefits.